### PR TITLE
feat(REST): Endpoint for Download Attachment Bundle of Component

### PR DESCRIPTION
**Pull request by tuannn2**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

- Run this Download Attachment Bundle of Release

Issue: #1955

### How To Test?
- Run command line: 
```
$ curl 'https://sw360.org/api/components/17653524/attachments/download' -i -X GET \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY4NDIwNTA1NSwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6ImdaWjBHaElRMGNLMjlwb0F0Vk56d05NWXBhQSIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.K9HK3NjDl6xHe3e5VUp7RYVUBHJg0tzuBW9XxsCD0DuyM3s7XtOXxKwce5qIHZXEAvMcuJG5Fpqh2fxQYhFxjxIgCX5H57s38yVZBpEcABjqjG9zvaaBGTVC9xXDXawhwlTbRYuRtnCRUBm7s8HFMGmdQDxPbeQueE78uLefJ-nJda3YfctKnPuXJ9rk2hKOVfuSy6YJrQG5J55x3v6GD_mSiQDGqrNAmrQgewIyeG2yJGw3UPZdMDDLbg1tXDEVqBuZyDBW0tEn4dEdV7LPjy9Zb9efTOvYukPLBAktwePeZ9NQxE2XVMxPLRmJdT6r7f5ly3DFpdMg7RSuPdfTBg' \
    -H 'Accept: application/*'
```
- Condition: Component already exists attachments. 

- Expected Output: File AttachmentBundle.zip

### CODE REVIEW

**Comment by nikkuma7** [comment]
Test successful.
image

**Review by afsahsyeda — CHANGES_REQUESTED** [review]

**Comment by tuannn2** [comment]
Thanks, @afsahsyeda 
I have fixed it. Please check it!

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ComponentSpecTest.java [hunk 1]
```diff
@@ -764,6 +764,16 @@ public void should_document_get_component_attachment() throws Exception {
                 .andDo(this.documentationHandler.document());
     }
 
+    @Test
+    public void should_document_get_component_attachment_bundle() throws Exception {
+        String accessToken = TestHelper.getAccessToken(mockMvc, testUserId, testUserPassword);
+        mockMvc.perform(get("/api/components/" + angularComponent.getId() + "/attachments/download")
+                        .header("Authorization", "Bearer " + accessToken)
+                        .accept("application/*"))
```
Comment (afsahsyeda) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ComponentSpecTest.java:772]: Please change this to "application/zip" as well

Comment (tuannn2) [reply rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ComponentSpecTest.java]: Thanks, Changes done.

**Review by afsahsyeda — APPROVED** [review]

**Pull request merged**